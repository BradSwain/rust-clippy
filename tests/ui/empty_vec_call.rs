#![warn(clippy::empty_vec_call)]

fn side_effect() -> usize {
    println!("side effect !");
    0
}

fn main() {
    let _a = [side_effect(); 0];
    let _v = vec![side_effect(); 0];

    // Should not emit warning because there are no side effects
    let _ok_a = [0; 0];
    let _ok_v = vec![0; 0];
}
