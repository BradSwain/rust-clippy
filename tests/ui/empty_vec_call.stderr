error: result of `side_effect()` is unused in empty array
  --> $DIR/empty_vec_call.rs:9:14
   |
LL |     let _a = [side_effect(); 0];
   |              ^^^^^^^^^^^^^^^^^^ help: consider using: `[] as [usize; 0]`
   |
   = note: `-D clippy::empty-vec-call` implied by `-D warnings`

error: result of `side_effect()` is unused in empty vec
  --> $DIR/empty_vec_call.rs:10:14
   |
LL |     let _v = vec![side_effect(); 0];
   |              ^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `vec![]`

error: aborting due to 2 previous errors

